### ISSUE TITLE
no-restyle: custom @utility classes reported as unrecognized even when all categories are allowed

### ISSUE THREAD

**Issue by wiesson** [shadcn-ui/lint#4]
A class declared with `@utility` in the project's Tailwind CSS is reported by `no-restyle` as unrecognized, although Tailwind generates it and `no-unknown-classes` accepts it:

```text
"shimmer" is not allowed on <Button>: the grammar does not recognize it. Fix the spelling, or use a class Tailwind generates.
```

The finding is emitted even when every category is allowed, so the only way out is listing each custom utility by name in `allow`. The suggested fix ("use a class Tailwind generates") is misleading for an agent, since the class is generated.

We hit this with `shimmer` from `shadcn/tailwind.css` (`@import "shadcn/tailwind.css"`, on a `MarkerContent` from the shadcn chat primitives); a local `@utility` reproduces it too.

## Reproduction

Paste into an empty directory (Node 20.19+, npm):

```bash
mkdir shadcn-lint-repro-utility && cd shadcn-lint-repro-utility
mkdir -p src/components/ui

cat > package.json <<'EOF'
{
  "private": true,
  "type": "module",
  "devDependencies": { "@shadcn/lint": "0.1.0", "oxlint": "1.81.0", "tailwindcss": "4.1.14" }
}
EOF
cat > components.json <<'EOF'
{
  "style": "new-york", "rsc": false, "tsx": true,
  "tailwind": { "config": "", "css": "src/app.css", "baseColor": "neutral", "cssVariables": true, "prefix": "" },
  "aliases": { "components": "@/components", "utils": "@/lib/utils", "ui": "@/components/ui", "lib": "@/lib", "hooks": "@/hooks" }
}
EOF
cat > tsconfig.json <<'EOF'
{ "compilerOptions": { "jsx": "react-jsx", "baseUrl": ".", "paths": { "@/*": ["./src/*"] } } }
EOF
cat > src/app.css <<'EOF'
@import "tailwindcss";

@utility shimmer {
  background-clip: text;
}
EOF
cat > src/components/ui/button.tsx <<'EOF'
export function Button({ className, ...props }: React.ComponentProps<"button">) {
  return <button className={["h-9 px-3 text-sm", className].join(" ")} {...props} />;
}
EOF
cat > src/page.tsx <<'EOF'
import { Button } from "@/components/ui/button";

export function Page() {
  return (
    <>
      <Button className="shimmer">Thinking…</Button>
      <div className="shimmer">Thinking…</div>
    </>
  );
}
EOF
cat > .oxlintrc.json <<'EOF'
{
  "jsPlugins": ["@shadcn/lint"],
  "rules": {
    "shadcn/no-restyle": ["warn", { "allow": ["layout", "typography", "color", "spacing", "shape", "effects", "motion"] }],
    "shadcn/no-unknown-classes": "warn"
  },
  "overrides": [{ "files": ["src/components/ui/**"], "rules": { "shadcn/no-restyle": "off" } }]
}
EOF

npm install --silent
npx oxlint src
```

Output:

```text
src/page.tsx:6:25: warning shadcn(no-restyle): "shimmer" is not allowed on <Button>: the grammar does not recognize it. Fix the spelling, or use a class Tailwind generates.
```

`no-unknown-classes` reports nothing for either element, so the class is known to the project's Tailwind.

## Expected

Custom utilities from the project's CSS (local `@utility` and utilities from imported CSS such as `shadcn/tailwind.css`) are classified, or at least treated like an allowed/unknown-but-valid class, instead of being reported as a spelling mistake.

## Where

[`rules/no-restyle.ts#L138-L152`](https://github.com/shadcn-ui/lint/blob/53de86f0e7dcc341a9cb45c383a9f2c454d1e958/packages/lint/src/rules/no-restyle.ts#L138-L152) emits `unclassifiedClass` for `verdict.category === "unclassified"` regardless of `allow`.

Workaround: `"allow": ["layout", …, "shimmer"]`.

## Environment

- `@shadcn/lint`: 0.1.0
- Oxlint: 1.81.0
- Tailwind CSS: 4.1.14 (real project: 4.3.3, shadcn 4.21.0)
- Node.js: 24.21.0
- npm (real project: pnpm 10.13.1), macOS
